User role resource errors when optional field `additional_attributes` is not defined.

## Describe the bug

When defining a resource to manage a user role in PrismaCloud, if the new optional field of `additional_attributes` is missing from the resource block, I get a index error. 

```
resource "prismacloud_user_role" "role_test_role_01" {
  name              = "test-role-01"
  description       = "Test Message"
  role_type         = "Account Group Read Only"
  account_group_ids = [prismacloud_account_group.ag_test_role_01.id]
}
```

## Expected behavior

Expected behavior would be for the user-role to be created in PrismaCloud.

## Current behavior

Stack trace from the terraform-provider-prismacloud_v1.1.4 plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 118 [running]:
github.com/terraform-providers/terraform-provider-prismacloud/prismacloud.parseUserRole(0xc0000a7260, 0x191c628)
        github.com/terraform-providers/terraform-provider-prismacloud/prismacloud/resource_user_role.go:140 +0x6c4
github.com/terraform-providers/terraform-provider-prismacloud/prismacloud.createUserRole(0xc0000a7260, 0x1b61300, 0xc000591790, 0x2, 0x25a5440)
        github.com/terraform-providers/terraform-provider-prismacloud/prismacloud/resource_user_role.go:198 +0x5b
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc000813830, 0xc000091ea0, 0xc00000c600, 0x1b61300, 0xc000591790, 0x1ac1d01, 0xc0004491f8, 0xc0002003f0)
        github.com/hashicorp/terraform-plugin-sdk@v1.9.0/helper/schema/resource.go:310 +0x365
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc0001f0000, 0xc000825a10, 0xc000091ea0, 0xc00000c600, 0xc00066c2c8, 0xc000180920, 0x1ac39a0)
        github.com/hashicorp/terraform-plugin-sdk@v1.9.0/helper/schema/provider.go:294 +0x99
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc00065e058, 0x1e6bf40, 0xc00018fd10, 0xc0000a6d90, 0xc00065e058, 0xc00018fd10, 0xc0004dab78)
        github.com/hashicorp/terraform-plugin-sdk@v1.9.0/internal/helper/plugin/grpc_provider.go:885 +0x8b4
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x1bced60, 0xc00065e058, 0x1e6bf40, 0xc00018fd10, 0xc0001234a0, 0x0, 0x1e6bf40, 0xc00018fd10, 0xc00033d080, 0x296)
        github.com/hashicorp/terraform-plugin-sdk@v1.9.0/internal/tfplugin5/tfplugin5.pb.go:3305 +0x217
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000782900, 0x1e761e0, 0xc000783500, 0xc000132e00, 0xc00080d500, 0x256b6a0, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.27.1/server.go:1024 +0x501
google.golang.org/grpc.(*Server).handleStream(0xc000782900, 0x1e761e0, 0xc000783500, 0xc000132e00, 0x0)
        google.golang.org/grpc@v1.27.1/server.go:1313 +0xd3d
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0005a8210, 0xc000782900, 0x1e761e0, 0xc000783500, 0xc000132e00)
        google.golang.org/grpc@v1.27.1/server.go:722 +0xa1
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.27.1/server.go:720 +0xa1

Error: The terraform-provider-prismacloud_v1.1.4 plugin crashed!

## Possible solution

A possible solution would be to not assume that there is a value at index position 1 for the additional_attributes element in the parseUserRole function:

```
func parseUserRole(d *schema.ResourceData) *role.Role {
	aspec := d.Get("additional_attributes").([]interface{})
	return &role.Role{
		Name:                    d.Get("name").(string),
		Description:             d.Get("description").(string),
		RoleType:                d.Get("role_type").(string),
		AccountGroupIds:         SetToStringSlice(d.Get("account_group_ids").(*schema.Set)),
		AssociatedUsers:         SetToStringSlice(d.Get("associated_users").(*schema.Set)),
		RestrictDismissalAccess: d.Get("restrict_dismissal_access").(bool),
		AdditionalAttributes: func() role.AdditionalAttributes {
			if len(aspec) == 0 {
				return role.AdditionalAttributes{}
			}
			aspecParsed := aspec[0].(map[string]interface{})
			return role.AdditionalAttributes{
				OnlyAllowCIAccess:      aspecParsed["only_allow_ci_access"].(bool),
				OnlyAllowComputeAccess: aspecParsed["only_allow_compute_access"].(bool),
				OnlyAllowReadAccess:    aspecParsed["only_allow_read_access"].(bool),
				HasDefenderPermissions: aspecParsed["has_defender_permissions"].(bool),
			}

		}(),
	}
}
```

## Steps to reproduce

You should be able to reproduce this issue by attempting to create a resource without specifying the `additional_attributes` argument in the resource block (similar to the resource block posted above).

--- a/prismacloud/resource_user_role.go
+++ b/prismacloud/resource_user_role.go
@@ -107,6 +107,7 @@ func resourceUserRole() *schema.Resource {
 				Type:        schema.TypeList,
 				Optional:    true,
 				Description: "Additional Parameters",
+				MaxItems: 1,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"only_allow_ci_access": {
@@ -137,21 +138,27 @@ func resourceUserRole() *schema.Resource {
 }
 
 func parseUserRole(d *schema.ResourceData) *role.Role {
-	aspec := d.Get("additional_attributes").([]interface{})[0].(map[string]interface{})
-	return &role.Role{
+	aspec := d.Get("additional_attributes").([]interface{})
+
+	ans := &role.Role {
 		Name:                    d.Get("name").(string),
 		Description:             d.Get("description").(string),
 		RoleType:                d.Get("role_type").(string),
 		AccountGroupIds:         SetToStringSlice(d.Get("account_group_ids").(*schema.Set)),
 		AssociatedUsers:         SetToStringSlice(d.Get("associated_users").(*schema.Set)),
 		RestrictDismissalAccess: d.Get("restrict_dismissal_access").(bool),
-		AdditionalAttributes: role.AdditionalAttributes{
-			OnlyAllowCIAccess:      aspec["only_allow_ci_access"].(bool),
-			OnlyAllowComputeAccess: aspec["only_allow_compute_access"].(bool),
-			OnlyAllowReadAccess:    aspec["only_allow_read_access"].(bool),
-			HasDefenderPermissions: aspec["has_defender_permissions"].(bool),
-		},
 	}
+
+	if len(aspec) > 0 {
+		if aspecs := aspec[0].(map[string]interface{}); len(aspecs) > 0 {
+			ans.AdditionalAttributes.OnlyAllowReadAccess = aspecs["only_allow_read_access"].(bool)
+			ans.AdditionalAttributes.OnlyAllowComputeAccess = aspecs["only_allow_compute_access"].(bool)
+			ans.AdditionalAttributes.OnlyAllowCIAccess = aspecs["only_allow_ci_access"].(bool)
+			ans.AdditionalAttributes.HasDefenderPermissions = aspecs["has_defender_permissions"].(bool)
+		}
+	}
+
+	return ans
 }
 
 func saveUserRole(d *schema.ResourceData, obj role.Role) {

